### PR TITLE
Update share-links.rst

### PR DESCRIPTION
**Pull request by Good-Soma**
Fixes 

-  #7430 

I added a passage that describe how deep link can be formatted, it value with an example that  clarify the message as a subsection in share-links.rst file  as you instructed.

I made the content to correlate with the previous note that had been written.

### CODE REVIEW

**Comment by Good-Soma** [comment]
Hello, 
I just sent a pull request that address the #7430 issue. Please, review it and merge it, if is okay, or notify me if I need to make any modification. Thanks

**Comment by Good-Soma** [comment]
> Hello @Good-Soma,
> 
> Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).
> 
> Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.
> 
> **Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?** Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.
> 
> This is a standard procedure for many open source projects.
> 
> Please let us know if you have any questions.
> 
> We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

/check-cla
I have signed the CLA in the website that I was directed to, but the check isn't turning green. I click on the detail beside the required box it showed, and downloaded the file in it, I wasn't able to open the file. Any hint?

**Comment by Good-Soma** [comment]
Okay, it has work.

File: source/collaborate/share-links.rst [hunk 1]
```diff
@@ -46,3 +46,32 @@ Share message links
 
   - Message previews respect channel membership permissions, so they’re only visible to users who have access to the original message. If the link is to a message in a public channel, any member of the team can see the message preview. If the link is to a message in a private channel or direct message, only members in that channel can see the message preview.
   - If you're unable to share links, contact your Mattermost system admin for assistance. An :doc:`SSL certificate (or a self-signed certificate) </onboard/ssl-client-certificate>` may be required for this functionality to work.
+
+What about **deep link**; how can it be configured?
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:50]:
```suggestion
Deep links
```

File: source/collaborate/share-links.rst [hunk 2]
```diff
@@ -46,3 +46,32 @@ Share message links
 
   - Message previews respect channel membership permissions, so they’re only visible to users who have access to the original message. If the link is to a message in a public channel, any member of the team can see the message preview. If the link is to a message in a private channel or direct message, only members in that channel can see the message preview.
   - If you're unable to share links, contact your Mattermost system admin for assistance. An :doc:`SSL certificate (or a self-signed certificate) </onboard/ssl-client-certificate>` may be required for this functionality to work.
+
+What about **deep link**; how can it be configured? 
+============================
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:51]:
```suggestion
--------------
```

File: source/collaborate/share-links.rst [hunk 3]
```diff
@@ -46,3 +46,32 @@ Share message links
...
+Deep link are unique link that has the ability to direct users to a specific resource on a web page or application, for example let's say you have an image on your website that you want your team to view, you can make them view the image and dissect the concept you want to share with them almost instantaneously with deep link. 
+
+Although you can treat the link you copy from your message and image, based on the illustration above as a deep link; hence, it takes collaborators to the object you wish to point them to; however, you will need to create a custom deep link if you want something like the opening and activation of Mattermost app on your mobile phone or web deep linking to specific resources through browser which may not be feasible through the link you copied from your messaging channels.
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:54]:
```suggestion
A Mattermost deep link is a URL that directs users to a specific location within Mattermost. Typically, these links are used to go to specific teams, channels, messages, or threads. 

.. tip::

  Deep links can also be used, in combination with bots, scripts, and integrations, to trigger specific actions within Mattermost.
```

File: source/collaborate/share-links.rst [hunk 4]
```diff
@@ -46,3 +46,32 @@ Share message links
...
   - If you're unable to share links, contact your Mattermost system admin for assistance. An :doc:`SSL certificate (or a self-signed certificate) </onboard/ssl-client-certificate>` may be required for this functionality to work.
+
+What about **deep link**; how can it be configured? 
+============================
+Deep link are unique link that has the ability to direct users to a specific resource on a web page or application, for example let's say you have an image on your website that you want your team to view, you can make them view the image and dissect the concept you want to share with them almost instantaneously with deep link. 
+
+Although you can treat the link you copy from your message and image, based on the illustration above as a deep link; hence, it takes collaborators to the object you wish to point them to; however, you will need to create a custom deep link if you want something like the opening and activation of Mattermost app on your mobile phone or web deep linking to specific resources through browser which may not be feasible through the link you copied from your messaging channels. 
+  
+To customize the Mattermost deep link for suitability:
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:56]:
```suggestion
Format deep links
```

File: source/collaborate/share-links.rst [hunk 5]
```diff
@@ -46,3 +46,32 @@ Share message links
...
+-----------------
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:57]:
```suggestion
~~~~~~~~~~~~~~~~~
```

File: source/collaborate/share-links.rst [hunk 6]
```diff
@@ -46,3 +46,32 @@ Share message links
...
+Which takes users directly to the content you have for them directly from your browser, follow the guidelines given to you from the previous section, which comprises: 
+Locate the permalink to the file. 
+Copying the URL 
+And distributing in any part of your message. 
+
+Furthermore, you can customize the link by building a button for it, thereby making it more attractive and interactive for your users or team. 
+
+Before you launch your link with assurance, test the link by yourself in your browser or mobile app to ascertain that it is readily available upon demand and compatible with the Mattermost functionary packages. 
+
+You are now acquainted with the techniques you can deploy to enhance the experience of your team in Mattermost collaborative application, by issuing straightforward reference within your content with deep links. Go ahead and add a deep link to your message.
```
Comment (cwarnermm) [inline source/collaborate/share-links.rst:77]:
```suggestion
Deep links must be formatted in Mattermost as follows:

- Deep link to a team: ``mattermost://<your-Mattermost-server-URL>/<team-name>``
- Deep link to a channel: ``mattermost://<your-Mattermost-server-URL>/<team-name>/channels/<channel-name>``
- Deep link to a message or thread: ``mattermost://<your-Mattermost-server-URL>/<team-name>/pl/<post-id>``
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @Good-Soma. I've shortened your docs PR proposal for brevity, readability, and technical accuracy.

**Comment by Rutam21** [comment]
@lindy65 Please assign this PR to me for QA Testing. Thanks.

**Comment by lindy65** [comment]
Assigned to you @Rutam21

**Comment by Good-Soma** [comment]
> Thanks, @Good-Soma. I've shortened your docs PR proposal for brevity, readability, and technical accuracy.

Okay, thanks I have seen it, plus my flaw, I will make the next document I will be working on better .  Notify me, if you want me to make further changes?

**Comment by Rutam21** [comment]
### The changes in this PR have been tested in the preview environment and all the required changes are available in the mentioned pages.

- The new `Deep Links` section has been added to the `Share links to channels and messages` page.
- This section is also listed in the list of contents on the right side of the page.

I've attached a video as confirmation.

https://github.com/user-attachments/assets/dd15448c-f88b-440b-b9d5-5ff8e0980024

cc: @lindy65

**Comment by lindy65** [comment]
Thanks @Rutam21 - counted towards Hacktoberfest!

cc @cwarnermm

**Pull request merged**